backend/server/public/src/admin.jsx

```diff
@@ -244,42 +237,93 @@ const Ad = ({ ad, onSuppressClick, onPermalinkClick }) => (
   </div>
 );
 
-let Ads = ({
-  ads,
-  onSuppressClick,
-  onKeyUp,
-  onPermalinkClick,
-  search,
-  pagination
-}) => (
-  <div id="ads">
-    <input
-      id="search"
-      placeholder="Search for ads"
-      onKeyUp={onKeyUp}
-      search={search}
-    />
-    {pagination ? <Pagination /> : ""}
-    {ads.map(ad => (
-      <Ad
-        ad={ad}
-        key={ad.id}
-        onSuppressClick={onSuppressClick}
-        onPermalinkClick={onPermalinkClick}
+class AdDetail extends React.Component {
+  constructor(props) {
+    super(props);
+  }
+
+  componentDidMount() {
+    console.log(this.props);
+    let ad_id = null;
+    if (this.props.match) {
+      ad_id = this.props.match.params.ad_id;
+    } else {
+      ad_id = this.props.requested_ad_id; // really state.permalinked_ad.requested_ad_id
+    }
+    store.dispatch(getOneAd(ad_id));
+  }
+
+  render() {
+    if (this.props.ads && this.props.ads[this.props.requested_ad_id].loaded) {
```

Comment:
We should make it so that `permalinked_ads.ads` always exists in the reducer. By providing a default state, that way we don't have to remember this check.